independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -361,6 +373,27 @@ void init() {
         return bindings;
     }
 
+    static Map<DotName, ClassInfo> findTransitiveInterceptorBindings(Collection<DotName> initialBindings,
+            IndexView index, Map<DotName, ClassInfo> result) {
+        // any annotation that contains interceptor binding annotation is effectively also an interceptor binding
```

Comment:
I don't think so. It must be annotated with `@InterceptorBinding`. From the spec:
_"An interceptor binding type may declare other interceptor bindings. Interceptor bindings are transitive — an interceptor binding declared by an interceptor binding type is inherited by all components and other interceptor binding types that declare that interceptor binding type."_

Reply:
So it seems that the spec indeed implies more strict rules whereas my implementation is lenient.
I'll revisit this on Mon and will try to alter it accordingly.